platforms/ide/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/parameters/BuildProgressListenerAdapter.java

```diff
@@ -1102,6 +1137,20 @@ private static List<Failure> toFailures(List<? extends InternalFailure> causes)
     }
 
     private static Failure toFailure(InternalFailure origFailure) {
+        if (origFailure instanceof InternalProblemAwareFailure) {
+            InternalProblemAwareFailure problemAwareFailure = (InternalProblemAwareFailure) origFailure;
+            List<InternalBasicProblemDetailsVersion3> problems = problemAwareFailure.getProblems();
+            List<ProblemReport> clientProblems = new ArrayList<>(problems.size());
+            for (InternalBasicProblemDetailsVersion3 problem : problems) {
+                clientProblems.add(createproblemReport(problem));
+            }
+            return new DefaultProblemAwareFailure(
+                origFailure.getMessage(),
+                origFailure.getDescription(),
+                toFailures(origFailure.getCauses()),
+                clientProblems
+            );
+        } else
         if (origFailure instanceof InternalTestAssertionFailure) {
```

Comment:
```suggestion
        } else if (origFailure instanceof InternalTestAssertionFailure) {
```

💅  Join the two lines.